xarray/core/dataarray.py

```diff
@@ -914,9 +914,13 @@ def encoding(self, value: Mapping[Any, Any]) -> None:
         self.variable.encoding = dict(value)
 
     def reset_encoding(self) -> Self:
+        warnings.warn("reset_encoding is deprecated, use `drop_encoding` instead")
```

Comment:
```suggestion
        warnings.warn("reset_encoding is deprecated since 2023.11, use `drop_encoding` instead")
```
Adding a date helps enforcing a switch I think.

Reply:
Good idea!